### PR TITLE
fix: vulnerability count message [HEAD-1024]

### PR DESCRIPTION
**Pull request by j-luong**
### Description

_This PR should switch to LS for vuln count and cleanup some redundant code._

### Checklist

- [x] Tests added and all succeed
- [x] Linted
- [ ] CHANGELOG.md updated
- [ ] README.md updated, if user-facing

### CODE REVIEW

File: src/snyk/snykOss/editor/editorDecorator.ts [hunk 1]
```diff
@@ -58,7 +57,7 @@ export class EditorDecorator {
           module.line - 1,
           this.editorLastCharacterIndex,
         ),
-        renderOptions: getRenderOptions(messages.vulnerabilityCount.fetchingVulnerabilities, this.themeColorAdapter),
+        renderOptions: getRenderOptions('', this.themeColorAdapter),
```
Comment (bastiandoetsch) [inline src/snyk/snykOss/editor/editorDecorator.ts:60]: question: Why did you remove the message?

File: src/snyk/snykOss/services/vulnerabilityCount/ossVulnerabilityCountService.ts [hunk 2]
```diff
@@ -35,23 +40,21 @@ export class OssVulnerabilityCountService implements Disposable {
...
     private readonly ossService: OssService,
     private readonly logger: ILog,
     private readonly editorDecorator: EditorDecorator,
     private readonly analytics: IAnalytics,
     private readonly configuration: IConfiguration,
   ) {}
-
   activate(): boolean {
     this.disposables.push(
-      (this.diagnostics = this.languages.createDiagnosticCollection(DIAGNOSTICS_OSS_COLLECTION_NAME)),
```
Comment (bastiandoetsch) [inline src/snyk/snykOss/services/vulnerabilityCount/ossVulnerabilityCountService.ts:48]: question: do we still create diagnostics for HTML somewhere?

File: src/snyk/snykOss/services/vulnerabilityCount/ossVulnerabilityCountService.ts [hunk 3]
```diff
@@ -35,23 +40,21 @@ export class OssVulnerabilityCountService implements Disposable {
...
-      this.workspace.onDidChangeTextDocument(ev => {
-        if (ev.contentChanges.length) {
-          this.processFile(ev.document);
+      this.workspace.onDidChangeTextDocument((ev: TextDocumentChangeEvent) => {
+        if (ev?.contentChanges.length) {
+          _.debounce(() => this.processFile(ev.document), 500);
```
Comment (bastiandoetsch) [inline src/snyk/snykOss/services/vulnerabilityCount/ossVulnerabilityCountService.ts:54]: suggestion: I think I'm seeing problems with the rendering of the decorator with this debounce. Can we consider removing it?

Comment (bastiandoetsch) [reply src/snyk/snykOss/services/vulnerabilityCount/ossVulnerabilityCountService.ts:54]: image

File: src/test/unit/snykOss/providers/vulnerabilityCountProvider.test.ts [hunk 4]
```diff
@@ -82,25 +110,44 @@ suite('OSS ModuleVulnerabilityCountProvider', () => {
...
+        } as unknown as InlineValueText['range'],
+      },
+    ];
+
+    sampelFileName = 'test.ts';
+    languageClientStub.sendRequest = sinon.stub().resolves(sampleInlineValueText);
 
     const count = await vulnerabilityCountProvider.getVulnerabilityCount(
-      'test.ts',
+      sampelFileName,
```
Comment (bastiandoetsch) [inline src/test/unit/snykOss/providers/vulnerabilityCountProvider.test.ts:142]: nitpick: typo

**Review by bastiandoetsch — APPROVED** [review]

File: src/snyk/snykOss/services/vulnerabilityCount/ossVulnerabilityCountService.ts (hunk 2 above)
Comment (j-luong) [reply src/snyk/snykOss/services/vulnerabilityCount/ossVulnerabilityCountService.ts:48]: No, it looks like this is needed to show HTML diagnostics, I've reverted this change

File: src/snyk/snykOss/editor/editorDecorator.ts (hunk 1 above)
Comment (j-luong) [reply src/snyk/snykOss/editor/editorDecorator.ts:60]: the change to the `this.workspace.onDidChangeTextDocument` workspace listener meant that this message did not update; as it's only briefly visible, I thought to leave it blank.

This can be reverted once the `this.workspace.onDidChangeTextDocument` listener is fixed.

File: src/snyk/snykOss/services/vulnerabilityCount/ossVulnerabilityCountService.ts (hunk 3 above)
Comment (j-luong) [reply src/snyk/snykOss/services/vulnerabilityCount/ossVulnerabilityCountService.ts:57]: hmm, this should not happen :). working on a fix.
